easybuild/easyblocks/i/imkl.py

```diff
@@ -234,6 +235,17 @@ def build_mkl_fftw_interfaces(self, libdir):
             for liball in glob.glob(os.path.join(interfacedir, '*', 'makefile')):
                 apply_regex_substitutions(liball, regex_nvc_subs)
 
+        # RPATH wrappers add -Wl,rpath arguments to all command lines, including when it is just compiling.
+        # The icx compiler by default warns about that, but the makefiles of the cdft wrappers set -Wall -Werror,
+        # which turns them into "linker input unused" errors:
+        # See https://github.com/easybuilders/easybuild-easyblocks/issues/2910
+        # Here, we patch the makefiles and add -Wno-unused-command-line-argument to avoid these warnings alltogether.
+        if get_software_root('intel-compilers') and build_option('rpath'):
+            if self.toolchain.options.get('oneapi') or self.toolchain.options.get('oneapi_c_cxx'):
+                regex_icx_subs = [('-Werror', '-Werror -Wno-unused-command-line-argument')]
```

Comment:
Maybe it makes more sense to replace `-Wall` than `-Werror`, since the order of options may matter.

When replacing `-Werror`, I think we assume that `-Wall` is listed earlier, so `-Wno-unused-*` can disable the warning.
When replacing `-Wall` to inject `-Wno-unused-*`, we're disabling that specific warning right after enabling all warnings (and before `-Werror` which enables all warnings to be treated as errors kicks in).

This may make this fix a bit more robust w.r.t. changes to the `makefile` being patched

```
regex_icx_subs = [('-Wall', '-Wall -Wno-unused-command-line-argument')]